File: code/1004. Max Consecutive Ones III.py
class Solution:
    def longestOnes(self, A: List[int], K: int) -> int:
        n = len(A)
        left = 0
        ans = 0
        zero = 0
        for right, val in enumerate(A):
            if val == 0:
                zero += 1
            while zero > K:
                if A[left] == 0:
                    zero -= 1
                left += 1
            ans = max(ans, right - left + 1)
        return ans

File: code/1004. Max Consecutive Ones III.cpp
class Solution {
public:
    int longestOnes(vector<int>& A, int K) {
        int n = A.size();
        int left = 0;
        int zero = 0;
        int ans = 0;
        for (int right = 0; right < n; right++) {
            if (A[right] == 0) zero++;
            while (zero > K) {
                if (A[left++] == 0) --zero;
            }
            ans = max(ans, right - left + 1);
        }
        return ans;
    }
};